site/templates/application/pixel.jet

```diff
@@ -0,0 +1,53 @@
+{{block pixelScripts(pixelID=config("meta_pixel_id"))}}
+  <!-- Meta Pixel Code -->
+  {{if pixelID != ""}}
+    <script>
+    function loadPixelScripts() {
+      !function(f,b,e,v,n,t,s)
+      {if(f.fbq)return;n=f.fbq=function(){n.callMethod?
+      n.callMethod.apply(n,arguments):n.queue.push(arguments)};
+      if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0';
+      n.queue=[];t=b.createElement(e);t.async=!0;
+      t.src=v;s=b.getElementsByTagName(e)[0];
+      s.parentNode.insertBefore(t,s)}(window, document,'script',
+      'https://connect.facebook.net/en_US/fbevents.js');
+
+      const pixelID = {{pixelID}};
+      return pixelID;
+    };
+    
+    </script>
+  {{end}}
+  <!-- End Meta Pixel Code -->
+
+  <script>
+    // Only send message to load Pixel if a/ the frontend cookie consent isn't required or b/ they've agreed to the cookie consent
+    var cookieConsentRequired = {{ isEnabled("frontend_cookie_consent") }};
+
+    document.addEventListener('s72loaded', function(event) {
+      let messagebus = event.detail.app.messagebus;
+      if(!cookieConsentRequired || event.detail.app.hasCookieConsent()) {
+        let pixelID = loadPixelScripts();
+        messagebus.publish('loaded-pixel-script', pixelID);
+      }
+      else {
+        // Listen for an 'accept' of the cookie consent, in which case send message to load Pixel
+        messagebus.subscribe('cookie-consent-set', function(event) {
+          let pixelID = loadPixelScripts();
+          messagebus.publish('loaded-pixel-script', pixelID);
```

Comment:
I reckon you should pass the messagebus into the loadPixelScripts function so you can publish in there instead of duplicating this line.

Reply:
good call. moved.